datafusion/core/src/physical_plan/sorts/cursor.rs

```diff
@@ -99,6 +100,16 @@ pub trait Cursor: Ord {
 
     /// Advance the cursor, returning the previous row index
     fn advance(&mut self) -> usize;
+
+    /// Go to reference row, returning the previous row index
+    fn seek(&mut self, row: usize) -> usize;
```

Comment:
Seek was added in anticipation of the TODO/planned optimizations.

a.k.a. cycles of slicing and concating of normalized key rows, also matching the cascaded sort recommendations in Graefe 2006.

Reply:
Used with latest changes.